server/middleware/validationMiddleware.ts

```diff
@@ -60,7 +59,9 @@ function validationMiddleware(conditionService: ConditionService, type?: new ()
       req.flash('validationErrors', JSON.stringify(flattenErrors(errors)))
       req.flash('formResponses', JSON.stringify(req.body))
 
-      return res.redirect('back')
+      const referer = req.get('Referer')
+      if (!referer) return next(new Error('Missing Referer header'))
+      return res.redirect(referer)
     } catch (error) {
       return next(
         new Error(`Failed to validate licence details for: ${req.params.nomisId}`, {
```

Comment:
mmmh.. going forward we might be able to remove this catch - this was in place as there was no error handling and it would cause an unhandled rejection due this requiring async operations

If express 5 handles these properly we could potentially just throw the error